python/ray/serve/_private/utils.py

```diff
@@ -659,8 +659,19 @@ def send_forever():
         self.pusher_thread.start()
 
     def __del__(self):
-        self.stop_event.set()
-        self.pusher_thread.join()
+        self.shutdown()
+
+    def shutdown(self):
+        """Shutdown metrics pusher gracefully.
+
+        This method will ensure idempotency of shutdown call.
+        """
+        if not self.stop_event.is_set():
+            self.stop_event.set()
+
+        if self.pusher_thread:
+            self.pusher_thread.join()
+            self.pusher_thread = None
```

Comment:
the conditional and setting the thread to `None` are unnecessary - `.join()` will return immediately on subsequent calls if the thread has already exited